tracer/src/Datadog.Trace/Iast/ReturnedHeadersAnalyzer.cs

```diff
@@ -0,0 +1,98 @@
+// <copyright file="ReturnedHeadersAnalyzer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Specialized;
+using System.Net;
+using System.Net.Mime;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+#if !NETFRAMEWORK
+using Microsoft.AspNetCore.Http;
+#endif
+using static Datadog.Trace.Telemetry.Metrics.MetricTags;
+
+#nullable enable
+namespace Datadog.Trace.Iast;
+
+internal static class ReturnedHeadersAnalyzer
+{
+    private const string ContentType = "content-type";
+    private const string XContentTypeOptions = "x-content-type-options";
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ReturnedHeadersAnalyzer));
+
+    // Analyze the headers. If the response is HTML, check for X-Content-Type-Options: nosniff. If it
+    // is not present, report a vulnerability. When getting the headers, make sure that keys are searched taking
+    // int account that can be case insensitive. Exclude vulnerability when return code is one of the ignorable.
+#if NETFRAMEWORK
+    internal static void Analyze(NameValueCollection responseHeaders, IntegrationId integrationId, string serviceName, int responseCode)
+#else
+    internal static void Analyze(IHeaderDictionary responseHeaders, IntegrationId integrationId, string serviceName, int responseCode)
+#endif
+    {
+        try
+        {
+            if (string.IsNullOrEmpty(serviceName) || IsIgnorableResponseCode((HttpStatusCode)responseCode))
+            {
+                return;
+            }
+
+            IastModule.OnExecutedSinkTelemetry(IastInstrumentedSinks.XContentTypeHeaderMissing);
+            string contentTypeValue = responseHeaders[ContentType];
+            string contentOptionValue = responseHeaders[XContentTypeOptions];
+
+            if (!IsHtmlResponse(contentTypeValue) || IsNoSniffContentOptions(contentOptionValue))
+            {
+                return;
+            }
+
+            LaunchXContentTypeOptionsVulnerability(integrationId, serviceName, contentTypeValue, contentOptionValue);
+        }
+        catch (Exception error)
+        {
+            Log.Error(error, $"{nameof(ReturnedHeadersAnalyzer)}.{nameof(Analyze)} exception");
+        }
```

Comment:
I think this will still prevent the `PlatformNotSupportedException` bubbling up right?

Reply:
That's what I was just thinking :) Done!